plugins/widget/plugin.js

```diff
@@ -3392,6 +3409,8 @@
 		widget.dragHandlerContainer = container;
 	}
 
+	var fakeParagraphs = [];
```

Comment:
Maybe renaming it into `dropLineRelation` or something like that would make more sense? And also refactoring it into an object:

```javascript
var dropLinePosition = {
    before: { position: CKEDITOR.LINEUTILS_BEFORE, element: beforeElement }
    after: { position: CKEDITOR.LINEUTILS_AFTER, element, afterElement }
}
```
you could use it as:

```javascript
moveDropLine( cell, dropLinePosition );
```
Also, I wouldn't clean up this object. You can reuse paragraphs as long as you want without the need to destroy/rebuilt them. Although it may require some testing if they can be easily garbage collected when detached from the DOM. However, it may be micro-optimization not worth the try.

Reply:
>Although it may require some testing if they can be easily garbage collected when detached from the DOM.

They won't be garbage collected when they are referenced in a variable which is accessible forever. We can't change it without heavy refactoring to whole plugin.

Reply:
`dropLinePosition`, `moveDropLine ` - I thought about better naming that `fakeParagraph` and didn't come up with anything. However your proposal doesn't sound right to me. I wouldn't rename `placeFakeParagraph` into `moveDropLine`.  This function doesn't really do anything to drop line, it appends temporary elements that are to be used for finding possible drop line positions, but still they might not be used for placing drop line when they are not the closests elements to mouse position.